tests/libcheckout.bats

```diff
@@ -95,6 +124,33 @@ teardown() {
   assert_output --partial "Revision: $SEMAPHORE_GIT_SHA not found .... Exiting"
 }
 
+@test "libcheckout - [PR] Checkout and use cache" {
+  export SEMAPHORE_GIT_REF_TYPE="pull-request"
+  export SEMAPHORE_GIT_REF="refs/pull/186/merge"
```

Comment:
We are using diffrent ref for PRs, in form of: `refs/semaphoreci/SEMAPHORE_GIT_SHA`, so here it will be `refs/semaphoreci/30774365e11f2b1e18706c9ed0920369f6d7c205`.

We are doing that, to be sure that we will fetch the same commit on each workflow/job run.